streamx-plugin/streamx-flink-kubernetes/src/main/scala/com/streamxhub/streamx/flink/kubernetes/KubernetesRetriever.scala

```diff
@@ -89,9 +90,19 @@ object KubernetesRetriever {
     val clientFactory: ClusterClientFactory[String] = clusterClientServiceLoader.getClusterClientFactory(flinkConfig)
     val clusterProvider: KubernetesClusterDescriptor = clientFactory.createClusterDescriptor(flinkConfig)
       .asInstanceOf[KubernetesClusterDescriptor]
-    val flinkClient: ClusterClient[String] = clusterProvider
-      .retrieve(flinkConfig.getString(KubernetesConfigOptions.CLUSTER_ID))
-      .getClusterClient
+
+    var flinkClient: ClusterClient[String] = null
+    Try {
+      flinkClient = clusterProvider
+        .retrieve(flinkConfig.getString(KubernetesConfigOptions.CLUSTER_ID))
+        .getClusterClient
+    } match {
+      case Success(v) => v
+      case Failure(e) =>
+        logError(s"Get flinkClient error.the error is:$e")
+        null
```

Comment:
Hi zhangjun, I think it would be better to throw the captured exception here. this is because the caller of KubernetesRetriever#newFinkClusterClient would do some special handling depending on the exception being thrown.

Reply:
such like that:

``` scala
    val flinkClient: ClusterClient[String] =
      Try(clusterProvider.retrieve(flinkConfig.getString(KubernetesConfigOptions.CLUSTER_ID)).getClusterClient)
        .failed
        .map(e => {
          logError(e.getMessage)
          throw e
        }).get
```